src/snowflake/snowpark/dataframe_stat_functions.py

```diff
@@ -449,21 +501,19 @@ def equal_condition_str(k: LiteralType) -> str:
                     "`seed` argument is ignored on `DataFrame` object. Save this DataFrame to a temporary table "
                     "to get a `Table` object and specify a seed.",
                 )
-            res_df = reduce(
-                lambda x, y: x.union_all(y, _emit_ast=False),
-                [
-                    self._dataframe.filter(col == k, _emit_ast=False).sample(
+
+            if self._dataframe._session.conf.get("use_simplified_query_generation"):
+                res_df = self._sample_by_with_percent_rank(col=col, fractions=fractions)
```

Comment:
does the output column name stay unchanged in the new approach?

Reply:
yes, sample_by will return a subset of rows which is specified by the input args. If you see the generated sql, we first add a new column to select a subset of rows, the do a `SELECT * EXCLUDE {PERCENT_RANK_COLUMN}`. This ensures that we preserve all the original columns in the dataframe and drop the singular column we added.

https://github.com/snowflakedb/snowpark-python/blob/780c28d5325a62fdb92bc2d5f27b3426f4f64ebb/src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py#L509